core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -446,6 +446,14 @@ default Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession sessio
         return Optional.empty();
     }
 
+    /**
+     * Get the physical metadata for a new table.
```

Comment:
physical -> effective

Also, a method name could be changed to reflect that.
Maybe `getNewTableEffectiveMetadata`